oauth2client/client.py

```diff
@@ -999,21 +999,19 @@ def _detect_gce_environment():
     #       could lead to false negatives in the event that we are on GCE, but
     #       the metadata resolution was particularly slow. The latter case is
     #       "unlikely".
-    connection = six.moves.http_client.HTTPConnection(
-        _GCE_METADATA_HOST, timeout=GCE_METADATA_TIMEOUT)
-
+    http = transport.get_http_object(timeout=GCE_METADATA_TIMEOUT)
+    headers = {_METADATA_FLAVOR_HEADER: _DESIRED_METADATA_FLAVOR}
     try:
-        headers = {_METADATA_FLAVOR_HEADER: _DESIRED_METADATA_FLAVOR}
-        connection.request('GET', '/', headers=headers)
-        response = connection.getresponse()
+        response, _ = transport.request(
+            http, _GCE_METADATA_URI, headers=headers)
         if response.status == http_client.OK:
```

Comment:
This can be structurally simplified to
```
return (
    response.status == http_client.OK and
    response.get(_METADATA_FLAVOR_HEADER) == _DESIRED_METADATA_FLAVOR)
```
. Maybe with shorter names for the constants it would look good laid out as well?